server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -194,6 +194,10 @@ private boolean syncSegments() {
                 indexShard.getReplicationTracker().isPrimaryMode(),
                 indexShard.state()
             );
+            if (indexShard.state() == IndexShardState.STARTED && indexShard.getEngine() instanceof InternalEngine) {
+                // Retrying to make sure that we do not lose this refresh event
+                return false;
+            }
```

Comment:
Is this part of the same bugfix or some other issue?

Reply:
It is part of the same bugfix. This only comes into picture when both primary as well as replica are getting initialized as part of restore. I will add a comment explaining this scenario.